### ISSUE TITLE
Segmentation Fault 

### ISSUE THREAD

**Issue by jaimef1603** [benzea/gnome-screencast#51]
I can't connect to my LG Smart TV  because gnome-network-displays crashes with a segmentation fault error. 
The output from ´G_MESSAGES_DEBUG=all gnome-network-displays --display=$DISPLAY´ is the one below. I'm running updated Arch and I have already installed openh264 but didn't solve it. 

``` 
(gnome-network-displays:27703): GLib-GIO-DEBUG: 14:36:21.674: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
(gnome-network-displays:27703): Gtk-DEBUG: 14:36:21.729: Connecting to session manager
(gnome-network-displays:27703): Gtk-DEBUG: 14:36:21.729: Failed to get the GNOME session proxy: The name org.gnome.SessionManager is not owned
(gnome-network-displays:27703): Gtk-DEBUG: 14:36:21.730: Failed to get the Xfce session proxy: The name org.xfce.SessionManager is not owned
** (gnome-network-displays:27703): DEBUG: 14:36:21.803: WFDP2PRegistry: Found a new device, creating provider
** (gnome-network-displays:27703): DEBUG: 14:36:21.803: WFDP2PProvider: Found a new sink with peer 0x55f0be5ec4e0 on device 0x55f0be9bdde0
** (gnome-network-displays:27703): DEBUG: 14:36:21.803: WfdP2PProvider: Discover is now set to 1
** (gnome-network-displays:27703): DEBUG: 14:36:21.803: SinkList: Adding a sink
** (gnome-network-displays:27703): DEBUG: 14:36:21.806: NdScreencastPortal: Aquired Portal proxy
** (gnome-network-displays:27703): DEBUG: 14:36:21.806: task is: 0x55f0be8cbbf0
** (gnome-network-displays:27703): DEBUG: 14:36:21.810: NdScreencastPortal: Received CreateSession response

** (gnome-network-displays:27703): WARNING **: 14:36:21.810: Error initing screencast portal: Failed to create portal session (response: 2)

** (gnome-network-displays:27703): WARNING **: 14:36:21.810: Falling back to X11! You need to fix your setup to avoid issues (XDG Portals and/or mutter screencasting support)!
** (gnome-network-displays:27703): DEBUG: 14:36:21.810: WFDP2PRegistry: Got NMClient
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.813: gdk_pixbuf_from_pixdata() called on:
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.814: 	Encoding raw
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.814: 	Dimensions: 16 x 16
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.814: 	Rowstride: 64, Length: 1048
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.814: 	Copy pixels == false
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.814: gdk_pixbuf_from_pixdata() called on:
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: 	Encoding raw
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: 	Dimensions: 16 x 16
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: 	Rowstride: 64, Length: 1048
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: 	Copy pixels == false
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: gdk_pixbuf_from_pixdata() called on:
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: 	Encoding raw
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: 	Dimensions: 16 x 16
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: 	Rowstride: 64, Length: 1048
(gnome-network-displays:27703): GdkPixbuf-DEBUG: 14:36:21.815: 	Copy pixels == false
** (gnome-network-displays:27703): DEBUG: 14:36:21.818: NdPulseaudio: Querying sink info by name
** (gnome-network-displays:27703): DEBUG: 14:36:21.819: NdPulseaudio: Error querying sink info
** (gnome-network-displays:27703): DEBUG: 14:36:21.819: NdPulseaudio: Got a sink info for the expected name
** (gnome-network-displays:27703): DEBUG: 14:36:24.568: Found openh264enc for video encoding.
** (gnome-network-displays:27703): DEBUG: 14:36:24.568: Found x264enc for video encoding.
** (gnome-network-displays:27703): DEBUG: 14:36:24.568: WFD: Did not find any usable AAC audio encoder!
** (gnome-network-displays:27703): DEBUG: 14:36:24.569: Got state change notification from streaming sink to state ND_SINK_STATE_WAIT_P2P
** (gnome-network-displays:27703): DEBUG: 14:36:24.571: WfdP2PProvider: Discover is now set to 0
** (gnome-network-displays:27703): DEBUG: 14:36:24.586: NdWfdP2PSink: Got P2P connection
** (gnome-network-displays:27703): DEBUG: 14:36:24.586: Found openh264enc for video encoding.
** (gnome-network-displays:27703): DEBUG: 14:36:24.586: Found x264enc for video encoding.
** (gnome-network-displays:27703): DEBUG: 14:36:24.586: WFD: Did not find any usable AAC audio encoder!
** (gnome-network-displays:27703): DEBUG: 14:36:24.586: Got state change notification from streaming sink to state ND_SINK_STATE_WAIT_SOCKET
** (gnome-network-displays:27703): DEBUG: 14:36:39.685: NdWfdP2PSink: Got client connection
** (gnome-network-displays:27703): DEBUG: 14:36:39.685: Got state change notification from streaming sink to state ND_SINK_STATE_WAIT_STREAMING
** (gnome-network-displays:27703): DEBUG: 14:36:40.198: WfdClient: OPTIONS querying done
** (gnome-network-displays:27703): DEBUG: 14:36:40.200: WFD query params
** (gnome-network-displays:27703): DEBUG: 14:36:40.292: WfdClient: GET_PARAMS done
** (gnome-network-displays:27703): DEBUG: 14:36:40.292: Add audio codec to params:
** (gnome-network-displays:27703): DEBUG: 14:36:40.292: WfdAudioCodec: LPCM, 3, latency: 0
** (gnome-network-displays:27703): DEBUG: 14:36:40.292: Add audio codec to params:
** (gnome-network-displays:27703): DEBUG: 14:36:40.292: WfdAudioCodec: AAC, 7, latency: 0
** (gnome-network-displays:27703): DEBUG: 14:36:40.293: Add audio codec to params:
** (gnome-network-displays:27703): DEBUG: 14:36:40.293: WfdAudioCodec: AC3, 7, latency: 0
** (gnome-network-displays:27703): DEBUG: 14:36:40.293: Add codec to params:
** (gnome-network-displays:27703): DEBUG: 14:36:40.293: WfdVideoCodec:
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * profile: 1
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * level: 8
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * native: 1920x1080 60p

** (gnome-network-displays:27703): WARNING **: 14:36:40.293: Resolution 29 not found in table VESA, but was in suppliments 0x3FFFFFFF

** (gnome-network-displays:27703): DEBUG: 14:36:40.293: Supported resolutions:
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 800x480 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 800x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 854x480 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 854x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 864x480 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 864x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 640x360 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 640x360 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 960x540 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.293:  * 848x480 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 640x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 720x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 720x480 60i
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 720x576 50p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 720x576 50i
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1280x720 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1280x720 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1920x1080 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1920x1080 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1920x1080 60i
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1290x720 25p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1280x720 50p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1920x1080 25p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1920x1080 50p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1920x1080 50i
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1280x720 24p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 1920x1080 25p
** (gnome-network-displays:27703): DEBUG: 14:36:40.294:  * 800x600 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 800x600 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1024x768 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1024x768 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1152x864 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1152x864 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1280x768 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1280x768 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1280x800 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1280x800 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1360x768 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1360x768 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1366x768 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1366x768 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1280x1024 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1280x1024 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1400x1050 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1400x1050 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1440x900 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1440x900 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1600x900 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1600x900 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.295:  * 1600x1200 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 1600x1200 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 1680x1024 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 1680x1024 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 1680x1050 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 1680x1050 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 1920x1200 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296: Add codec to params:
** (gnome-network-displays:27703): DEBUG: 14:36:40.296: WfdVideoCodec:
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * profile: 2
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * level: 8
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * native: 1920x1080 60p

** (gnome-network-displays:27703): WARNING **: 14:36:40.296: Resolution 29 not found in table VESA, but was in suppliments 0x3FFFFFFF

** (gnome-network-displays:27703): DEBUG: 14:36:40.296: Supported resolutions:
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 800x480 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 800x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 854x480 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 854x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 864x480 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 864x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 640x360 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.296:  * 640x360 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 960x540 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 848x480 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 640x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 720x480 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 720x480 60i
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 720x576 50p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 720x576 50i
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1280x720 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1280x720 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1920x1080 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1920x1080 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1920x1080 60i
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1290x720 25p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1280x720 50p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1920x1080 25p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1920x1080 50p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1920x1080 50i
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1280x720 24p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1920x1080 25p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 800x600 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 800x600 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.297:  * 1024x768 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1024x768 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1152x864 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1152x864 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1280x768 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1280x768 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1280x800 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1280x800 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1360x768 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1360x768 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1366x768 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1366x768 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1280x1024 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1280x1024 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1400x1050 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1400x1050 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1440x900 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1440x900 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1600x900 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1600x900 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1600x1200 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1600x1200 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1680x1024 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.298:  * 1680x1024 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.299:  * 1680x1050 30p
** (gnome-network-displays:27703): DEBUG: 14:36:40.299:  * 1680x1050 60p
** (gnome-network-displays:27703): DEBUG: 14:36:40.299:  * 1920x1200 30p

** (gnome-network-displays:27703): WARNING **: 14:36:40.299: WfdClient: No resolution found, falling back to standard FullHD resolution.
** (gnome-network-displays:27703): DEBUG: 14:36:40.299: selected resolution 1920, 1080 @30
** (gnome-network-displays:27703): DEBUG: 14:36:40.301: WfdClient: SET_PARAMS done
** (gnome-network-displays:27703): DEBUG: 14:36:40.305: WfdClient: Initialization done!
** (gnome-network-displays:27703): DEBUG: 14:36:40.306: use x11: 1
[1]    27703 segmentation fault (core dumped)  G_MESSAGES_DEBUG=all gnome-network-displays --display=$DISPLAY
```

**Comment by benzea** [comment]
Odd, are you able to get a backtrace of the crash? (using `coredumpctl gdb gnome-network-displays`  maybe or by running it again in gdb?)

**Comment by jaimef1603** [comment]
Yes, this is the output when running `coredumpctl gdb gnome-network-displays`
```
       Message: Process 27703 (gnome-network-d) of user 1000 dumped core.
                
                Stack trace of thread 27725:
                #0  0x00007f691b7f4475 g_mutex_lock (libglib-2.0.so.0)
                #1  0x00007f691af5d126 n/a (libgstreamer-1.0.so.0)
                #2  0x00007f691af5a0d6 gst_bin_add (libgstreamer-1.0.so.0)
                #3  0x000055f0bdbf9698 n/a (gnome-network-displays)
                #4  0x00007f691a4756d0 ffi_call_unix64 (libffi.so.6)
                #5  0x00007f691a4750a0 ffi_call (libffi.so.6)
                #6  0x00007f691b93644e g_cclosure_marshal_generic (libgobject-2.0.so.0)
                #7  0x00007f691b92ccaa g_closure_invoke (libgobject-2.0.so.0)
                #8  0x00007f691b91a706 n/a (libgobject-2.0.so.0)
                #9  0x00007f691b91de2c g_signal_emit_valist (libgobject-2.0.so.0)
                #10 0x00007f691b91f5cb g_signal_emit_by_name (libgobject-2.0.so.0)
                #11 0x000055f0bdbfd954 n/a (gnome-network-displays)
                #12 0x00007f691a4756d0 ffi_call_unix64 (libffi.so.6)
                #13 0x00007f691a4750a0 ffi_call (libffi.so.6)
                #14 0x00007f691b936f18 g_cclosure_marshal_generic_va (libgobject-2.0.so.0)
                #15 0x00007f691b91ea5a g_signal_emit_valist (libgobject-2.0.so.0)
                #16 0x00007f691b91f700 g_signal_emit (libgobject-2.0.so.0)
                #17 0x000055f0bdc03f28 n/a (gnome-network-displays)
                #18 0x00007f691a4756d0 ffi_call_unix64 (libffi.so.6)
                #19 0x00007f691a4750a0 ffi_call (libffi.so.6)
                #20 0x00007f691b936f18 g_cclosure_marshal_generic_va (libgobject-2.0.so.0)
                #21 0x00007f691b91ea5a g_signal_emit_valist (libgobject-2.0.so.0)
                #22 0x00007f691b91f700 g_signal_emit (libgobject-2.0.so.0)
                #23 0x000055f0bdc02bf5 n/a (gnome-network-displays)
                #24 0x00007f691a91c014 n/a (libgstrtspserver-1.0.so.0)
                #25 0x000055f0bdc0285f n/a (gnome-network-displays)
                #26 0x00007f691a91b9f4 gst_rtsp_media_factory_construct (libgstrtspserver-1.0.so.0)
                #27 0x00007f691a930a6b n/a (libgstrtspserver-1.0.so.0)
                #28 0x00007f691a9318e6 n/a (libgstrtspserver-1.0.so.0)
                #29 0x00007f691a933cb3 n/a (libgstrtspserver-1.0.so.0)
                #30 0x00007f691a9372fb gst_rtsp_client_handle_message (libgstrtspserver-1.0.so.0)
                #31 0x00007f691a96cafa n/a (libgstrtsp-1.0.so.0)
                #32 0x00007f691b841bb0 g_main_context_dispatch (libglib-2.0.so.0)
                #33 0x00007f691b843b11 n/a (libglib-2.0.so.0)
                #34 0x00007f691b844a63 g_main_loop_run (libglib-2.0.so.0)
                #35 0x00007f691a90e042 n/a (libgstrtspserver-1.0.so.0)
                #36 0x00007f691b819cf7 n/a (libglib-2.0.so.0)
                #37 0x00007f691b820a31 n/a (libglib-2.0.so.0)
                #38 0x00007f691a48557f start_thread (libpthread.so.0)
                #39 0x00007f691a82e0e3 __clone (libc.so.6)
                
                Stack trace of thread 27704:
                #0  0x00007f691a823667 __poll (libc.so.6)
                #1  0x00007f691b843a80 n/a (libglib-2.0.so.0)
                #2  0x00007f691b843b51 g_main_context_iteration (libglib-2.0.so.0)
                #3  0x00007f691b843ba2 n/a (libglib-2.0.so.0)
                #4  0x00007f691b820a31 n/a (libglib-2.0.so.0)
                #5  0x00007f691a48557f start_thread (libpthread.so.0)
                #6  0x00007f691a82e0e3 __clone (libc.so.6)
                
                Stack trace of thread 27705:
                #0  0x00007f691a823667 __poll (libc.so.6)
                #1  0x00007f691b843a80 n/a (libglib-2.0.so.0)
                #2  0x00007f691b844a63 g_main_loop_run (libglib-2.0.so.0)
                #3  0x00007f691b9b0a38 n/a (libgio-2.0.so.0)
                #4  0x00007f691b820a31 n/a (libglib-2.0.so.0)
                #5  0x00007f691a48557f start_thread (libpthread.so.0)
                #6  0x00007f691a82e0e3 __clone (libc.so.6)
                
                Stack trace of thread 27703:
                #0  0x00007f691a823667 __poll (libc.so.6)
                #1  0x00007f691b843a80 n/a (libglib-2.0.so.0)
                #2  0x00007f691b843b51 g_main_context_iteration (libglib-2.0.so.0)
                #3  0x00007f691b9f35ce g_application_run (libgio-2.0.so.0)
                #4  0x000055f0bdbf90ef main (gnome-network-displays)
                #5  0x00007f691a758ee3 __libc_start_main (libc.so.6)
                #6  0x000055f0bdbf913e n/a (gnome-network-displays)

GNU gdb (GDB) 8.3
Copyright (C) 2019 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/gnome-network-displays...
(No debugging symbols found in /usr/bin/gnome-network-displays)

warning: core file may not match specified executable file.
[New LWP 27725]
[New LWP 27704]
[New LWP 27705]
[New LWP 27703]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
Core was generated by `gnome-network-displays --display=:0'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f691b7f4475 in g_mutex_lock () from /usr/lib/libglib-2.0.so.0
[Current thread is 1 (Thread 0x7f691574e700 (LWP 27725))]
```

And this is the output of gdb when running it again from there: 

```
(gdb) continue
Continuing.
[New Thread 0x7f0c7d7ad700 (LWP 1555)]

Thread 4 "pool-gnome-netw" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7f0c7d7ad700 (LWP 1555)]
0x00007f0c83853475 in g_mutex_lock () from /usr/lib/libglib-2.0.so.0
(gdb) continue
Continuing.
[Thread 0x7f0c7d7ad700 (LWP 1555) exited]
[Thread 0x7f0c7ec45700 (LWP 1463) exited]
[Thread 0x7f0c7f446700 (LWP 1462) exited]

Program terminated with signal SIGSEGV, Segmentation fault.
The program no longer exists.
```

**Comment by jaimef1603** [comment]
It happens when it reaches the connection with the TV, it connects for 1 or 2 seconds and then crashes.

**Comment by benzea** [comment]
Yeah, it is the point where the pipeline is created. You are lacking the debugging symbols unfortunately, but it is a bug must be when the video source is created (a backtrace from GDB should show a lot more, not just the function list).

Check whether you have `ximagesrc` and `intervideosink` available using `gst-inspect-1.0`.

**Comment by jaimef1603** [comment]
Yeah, I guess that was the problem, i dind't have them installed, I have installed them and the segmentation fault has gone. Now I have another issue very similar to #49 and #20, with this error message: 
`Warning from inter video sink: Pipeline construction is invalid, please add queues.`
